src/main/java/uk/ac/sanger/sccp/stan/service/work/WorkServiceImp.java

```diff
@@ -193,6 +192,41 @@ public Work link(Work work, Collection<Operation> operations) {
         return workRepo.save(work);
     }
 
+    @Override
+    public void link(Collection<Work> works, Collection<Operation> operations) {
+        if (operations.isEmpty() || works.isEmpty()) {
+            return;
+        }
+        if (works.size()==1) {
+            link(works.iterator().next(), operations);
+            return;
```

Comment:
Sorry, I didn't this. Why this check  if there is only one work number?

Reply:
`works.size()==1` is likely to be a very common use case. (I expect that nearly every time someone provides work numbers for this, there will be only one of them.) So in that case I fall back to one-work method, which is simpler and likely to be faster.

Reply:
Cool 👍